starknet_py/net/account/account.py

```diff
@@ -96,6 +99,10 @@ def __init__(
     def address(self) -> int:
         return self._address
 
+    @property
+    def cairo_version(self) -> int:
+        return self._cairo_version
```

Comment:
here you could do instead something like

```
@property
async def cairo_version(self) -> int:
    if self._cairo_version is None:
        # Get the class hash of self.address
        class_ = await self._client.get_class_at(self.address)
        self.cairo_version = 1 if isinstance(class, Sierra) else 0
    return self._cairo_version
```

Reply:
I'm not sure if an async property would be a good UI. However, we've got a sync interface for that, so something like this would work without `await`ing:
```py
@property
def cairo_version(self) -> int:
    if self._cairo_version is None:
        if isinstance(self._client, GatewayClient):
            contract_class = self._client.get_full_contract_sync(contract_address=self._address)
        else:
            contract_class = self.client.get_class_at_sync(contract_address=self._address)
        self._cairo_version = 1 if isinstance(contract_class, SierraContractClass) else 0
    return self._cairo_version
```
wdyt?

Reply:
well, just see this comment now. Async property you're right may not have been the best thing, but with sync definitely as It's quite common to use property to cache a given computed/fetched data

there is no information regarding the internal implementation for the outside world, use you still use `self.cairo_version`